### DISCUSSION TITLE
Question/discussion about emulating roomtemp for rego2000/3000(bosch 3800i)

### DISCUSSION THREAD

**Discussion by mattfro** [emsesp/EMS-ESP32#1691] [General]
What should I choose for control?
I have tried RC100, nothing show up(in ems-esp/devices, if I choose RC100H it shows up but I get an alarm and the temperature is never shown in the pump.
If I chose RC200, it shows up in devices and temperature is set in the pump the first time, but if I update the temperature it does not update in emulated RC200 or pump. I can get it sometimes to update if I set control again and remotetemp. But not working always.

not sure if this is supported for this pump, I think Bosch CR10/RC100 is supported as a real one and also Bosch CR5 RF. 
So I'm not sure if I have understood it wrongly, or configured it wrong or it does not work.

And if you want logs, what should I test with, that should most likely work with my "pump", so we can go from there. I can then write a issue ticket if you like? But first, it's more like a question if I'm even understanding or doing it correctly.

**Comment by MichaelDvP** [comment]
Please check what options the boiler have for control and if they match the emsesp entity selection of thermostat/hc1/control. 
You need to set control first to RC100(H) or RC200, and after this is active you can send remotetemp. Emsesp uses the control setting to decide if RC200 or RC100H is emulated, in other positions the remottemp is not set. If the control setting does not match the boiler option (e.g emsesp setting RC200, but boiler shows CR10) you will run into errors.

**Comment by mattfro** [comment]
Well...the problem is, boiler/heatpump itself does not have any settings or does not show what kind of "sensor" you have. And the only one I did not get any errors in the pump itself and it got a value that stays the same was rc200. But I wonder if I need to do full reset on the pump to get those settings. I have been through all settings including service settings. Can't find anything about the roomtemperature sensor. With RC200 I can see the value of the remotetemp. But after that it never gets a new value, so if I set it as 25C it will stay as that even if I change it in ems-esp. Maybe I need to test more...perhaps take some logs. For me it's really difficult to understand all ems logs. 

And btw. when I get that value once from the ems-esp I can also see the "room temperature offset setting" in the UI, so I can also change that manually there(in the pump). 

Settings that I have found from a manual, is on the sensor side: CR10 for example you need to set to Fb (remote control) mode and use HC1.  

What I know 3800i should support CR10 and CR5 RF at least. 
CR5 RF only needs to be connected and that's it. No settings at all according to the manual, only pair the sensor to mx200(which is the receiver).

Seems, I need to do some more testing :) if you don't have any other ideas?

**Reply by MichaelDvP** [reply]
> itself does not have any settings or does not show what kind of "sensor" you have

**Again:** You have to configure in the thermstat that there is a remote of type RC200 or RC100H, and this shows in emsesp as thermostat/hc1/control. If this setting is wrong, **it can not work!** I will add some code to switch off the remote temp on wrong settings.
> What I know 3800i should support CR10 and CR5 RF at least.

If you can provide a full log of the real remotes, we can try to emulate. We have no CR10 and no CR5FRF in the database yet, so no user have these devices.

**Reply by mattfro** [reply]
Ok... will try again. Thanks, CR10 should be same as RC100, without the H I think. So if I can choose RC100, it should work :) But that is not possible to choose, only RC100H (Does H stand for humidity?) Strange tat RC200 was working the closest when I tried.
Anyways, will get back when I test more.

I do not have any CR10 or RC100 or even CR5RF :( That's why I want to emulate it instead. Wasn't there a log for CR10 in some old discussion or trouble report?

**Comment by mattfro** [comment]
RC200 and I got the value in my pump
![image](https://github.com/emsesp/EMS-ESP32/assets/46936922/b3c965b3-3811-47eb-aa2d-6f8f9b8a5fbc)

here are logs also:
[rc200_1.log](https://github.com/emsesp/EMS-ESP32/files/14952206/rc200_1.log)

[K000+00:05:09.437 I 190: [command] [0;33mCalling command 'thermostat/control' (control device) with value RC200
[K000+00:07:47.941 I 1165: [command] [0;33mCalling command 'thermostat/control' (control device) with value RC200
[K000+00:07:50.212 I 1178: [command] [0;33mCalling command 'thermostat/remotetemp' (room temperature from remote) with value 23

seems I needed to set it twice...and when I got it working, seems after that I can't update the values anymore.
I don't know if you have time to check or so... and in the pump I checked many time, no way to change type of thermostat there in the UI.
Seems for older version, like bosch compress 3000 there is.  So I don't know if they have kind of made it automatic or something to figure out what type of sensor it is. 

I really would not like to buy a CR10 or CR5RF. But I guess, I'm going to need to do so if this wont work. 

Going tomorrow, or today take logs with RC100H...I could see it that the pump, could just get a line there with no temp.

**Reply by MichaelDvP** [reply]
Changing the control does not work on your system:
` Me(0x0B) -W-> thermostat(0x10), HPMode(0x0291), data: 01 (offset 3)`
`thermostat(0x10) -W-> Me(0x0B), HPMode(0x0291), data: 01 00 00 00 00 00`
offset 3 is still 00

I've made a test build to emulate RC200 if control is set to RC310. We can also try with CR10, i've found infos for CR10 in #826, it's identical with RC100. Don't know if there is another setting needed for remote temperature on your system. I see that husdata get it managed, but H66 is closed source, no way to look at it.

Check with https://github.com/MichaelDvP/EMS-ESP32/releases/tag/test

**Comment by mattfro** [comment]
Tested it...seem when I set temperature I get several RC200 and it does not work getting also alarm 3071.
[testbuild1.log](https://github.com/emsesp/EMS-ESP32/files/14955559/testbuild1.log)

**Reply by MichaelDvP** [reply]
Ups. i fixed the multiple remotes and added a RC100, no need to set thermostat/hc1/control, just send thermostat/hc1/remotetemp to create a RC100. Please update again.

**Comment by mattfro** [comment]
ok, a bit better now. But no dice :)
in ems-esp it looks ok... pump I see only --C on the indoor temp and have alarm 3091 seems to come and go.
[testbuild2.log](https://github.com/emsesp/EMS-ESP32/files/14956616/testbuild2.log)
![image](https://github.com/emsesp/EMS-ESP32/assets/46936922/b5fa8fd0-50f4-4025-bd07-a1c687ff8f8a)

**Reply by MichaelDvP** [reply]
Hm, the error (0xBF) comes 40 sec after the latest remote message, maybe the cable remotes sending more often and timeout is shorter. I set repeat rate to 15 sec now.

**Comment by mattfro** [comment]
[testbuild3.log](https://github.com/emsesp/EMS-ESP32/files/14957500/testbuild3.log)
what are those incomplete messages? Seems we start to get those more when activating remotetemp

**Reply by mattfro** [reply]
Same result otherwise... btw, I do wonder why with setting to RC200 was the only time I actually or the pump could actually get a value for the remotetemperature form ems-esp. Is the temperature value sent in wrong format or something(or wrong offset)?

**Reply by mattfro** [reply]
one other thing... the manuals on the sensor, it says set the sensor to Fb mode(control mode) could this be something that also needs to be sent? What ever that means, otherwise you will get an alarm. 
![image](https://github.com/emsesp/EMS-ESP32/assets/46936922/89f6590c-b021-4c9c-9ac1-2bb1477a27a9)

**Reply by MichaelDvP** [reply]
This is done by device address (0x38 for remote, 0x18 for master) and telegram (remote sends to master with different telegram, not to boiler). 
But for all other master thermostats there is a setting to accept the remote. Also i do not see any queries to the remote in the log.

I've changed the default (without control setting) to RC200, check it.

**Comment by mattfro** [comment]
OMG this works....but not 100% sure if it fully works, but it updates the temperature even on the pump from 22 to 23% on the remotetemp
[rc200_testbuild4.log](https://github.com/emsesp/EMS-ESP32/files/14958124/rc200_testbuild4.log)

so far no alarm, and it updates the room temperature information on the pump

**Reply by mattfro** [reply]
Getting still quite many Incomplete messages:
000+00:12:06.128 W 1290: [telegram] Incomplete Rx: 09 B8 00 F4
000+00:12:06.552 W 1291: [telegram] Incomplete Rx: 09 B8 00 F4
000+00:12:13.489 W 1292: [telegram] Incomplete Rx: 09 B8 40 89
000+00:12:23.499 W 1293: [telegram] Incomplete Rx: B8 00 E4 00 10 20 30 48 00 CB 16 01 18 00 00 00 00 02 B2 00 00 01 16 00 00 00 00 01 1F 01 19
000+00:12:46.127 W 1294: [telegram] Incomplete Rx: 09 B8 00 F4
000+00:13:06.489 W 1295: [telegram] Incomplete Rx: B8 00 F7 00 D1 00 00 07
000+00:13:10.464 W 1296: [telegram] Incomplete Rx: B8 00 F7 00 D1 00 00 07
000+00:13:12.463 W 1297: [telegram] Incomplete Rx: B8 00 F7 00 D1 00 00 07
000+00:13:26.476 W 1298: [telegram] Incomplete Rx: B8 00 F7 00 D1 00 00 07
000+00:13:29.426 W 1299: [telegram] Incomplete Rx: B8 00 F7 00 D1 00 00 07
000+00:13:35.278 W 1300: [telegram] Incomplete Rx: 09 B8 40 89
000+00:13:35.476 W 1301: [telegram] Incomplete Rx: B8 00 F7 00 D1 00 00 07
 
 why do they popup when adding remotetemp?

**Reply by MichaelDvP** [reply]
Looks like a timing issue with the poll-acks of the remote thermostat. Can you check a different tx-mode, i think HT3 should work better,but also try the others.

**Reply by mattfro** [reply]
Anyways great job, you're my hero now :D
Seems to work fine for all the time it has been on now, except those. I had it on ems+, trying HT3 now.

HT3 seems worse:
000+00:16:19.310 E 114: [telegram] Last Tx Read operation failed after 3 retries. Ignoring request: 8B 90 FF 00 19 01 A5
000+00:16:27.225 E 115: [telegram] Last Tx Read operation failed after 3 retries. Ignoring request: 8B 90 FF 00 19 01 A6
000+00:16:35.226 E 116: [telegram] Last Tx Read operation failed after 3 retries. Ignoring request: 8B 90 FF 00 19 03 72
000+00:16:49.262 E 117: [telegram] Last Tx Read operation failed after 3 retries. Ignoring request: 8B 90 FF 00 19 03 68
000+00:17:52.226 E 118: [telegram] Last Tx Read operation failed after 3 retries. Ignoring request: 8B 88 FF 00 19 03 A2
000+00:17:59.236 E 119: [telegram] Last Tx Read operation failed after 3 retries. Ignoring request: 8B 88 FF 00 19 03 88
000+00:18:07.237 E 120: [telegram] Last Tx Read operation failed after 3 retries. Ignoring request: 8B 90 FF 00 19 01 A5
000+00:18:15.273 E 121: [telegram] Last Tx Read operation failed after 3 retries. Ignoring request: 8B 90 FF 00 19 01 A6
000+00:18:25.237 E 122: [telegram] Last Tx Read operation failed after 3 retries. Ignoring request: 8B 90 FF 00 19 03 72


And if I reboot ems-esp it will forget all about RC200 :)

**Reply by mattfro** [reply]
I'm a bit concern about those messages...there are a lot of them. ems, ems+ and ht3 is worse with much more strange read operations.
Possible to improve it somehow? Need more logs? :)

**Reply by MichaelDvP** [reply]
I've tried to make the break timing a bit better, maybe it helps. Please try my new dev build. 
What isthe difference between the tx-modes? Ok HT3 givesTX errors, but what about ems/ems+/hardware?
Please send logs> 2min with ems and ems+ timings. Maybe there is a difference giving a hint whatis wrong.
I also like to know what the F7 requests to remote are, this should be displayed in the new dev.

**Reply by MichaelDvP** [reply]
I've made some tests with my ems-boiler. Without poll-ack i have no incomplete anymore and with 15 sec repeat rate also no errors.
I've update my build, please check if this works for your system too.

**Reply by mattfro** [reply]
hmm... build with this one?
https://github.com/MichaelDvP/EMS-ESP32/commit/4d400bd7cede5455cf17880d3975b3a85192ae5c: no poll-ack for remote thermostat (MichaelDvP)

Does not seem to work at all, now the pump does not see or notice that I have a room temp sensor.
[rc200_testbuild5.log](https://github.com/emsesp/EMS-ESP32/files/14976180/rc200_testbuild5.log)

**Reply by mattfro** [reply]
Actually, it seems to come and go. I sometimes see it and then it's gone.

**Reply by MichaelDvP** [reply]
Ok, i've added back the poll-ack. 
The controller often requests F7 for Version telegram, i've added a reply for that, may it helps.
You have also some incomplete in this version without poll-ack. They are not induced by remote emulation. 
How is the ems-esp connected? 
(bus/service-jack? cable diameter and length? If bus connect - split or output of a module?

**Reply by mattfro** [reply]
I use this from bbqkees:
Gateway S3 (Standard WiFi Edition V3)
with EMS service cable 110cm

straight connection and no other things connected.

**Comment by mattfro** [comment]
New log with your latest build:
[rc200_testbuild6.log](https://github.com/emsesp/EMS-ESP32/files/14978125/rc200_testbuild6.log)

Maybe those incomplete does not do any harm in the end? or...

**Reply by MichaelDvP** [reply]
A few incompletes do no harm, most messages are repeated often, so ems-esp do not miss the information. 
The first part of the log with nearly ervery second an incomplete is too much. but in the full log with a few per minute it's ok.
The bus cables are unshielded and can be disturbed, Bosch recommends to seperate power-cables and bus-cables when installing.

Example for repeated messages: the boiler sends very often the info, that D1-telegram have 3 data-bytes: `boiler(0x08) -B-> All(0x00), MenuConfig(0xF7), data: D1 00 00 07` and some of these gives error, because the 2 first bits are disturbed: `Incomplete Rx: B8 00 F7 00 D1 00 00 07` (fist byte should be 88). But i think i's only inside ems-esp-receiver and not on the bus. 
BTW: we know only first byte of D1 as outdoor-temperature, could you `read 8 D1` to see the other unknown bytes. 

I like to suppress the lot of `controller(0x09) -R-> thermostat(0x38), MenuConfig(0xF7), length: 0x02, data: 02 00 00` and added an answer to these requests, maybe it works. Please check.

**Reply by mattfro** [reply]
Ok...it's a bit tricky to get an power cable there. Not impossible but would not like to have it if not really needed.
And I also, if I don't use remotetemp I don't see those.

Anyways here is the readout:
ems-esp:# read 8 D1
000+19:18:03.718 N 6: [emsesp] boiler(0x08) -W-> Me(0x0B), UBAOutdoorTemp(0xD1), data: 00 34 00

**Reply by mattfro** [reply]
> I like to suppress the lot of `controller(0x09) -R-> thermostat(0x38), MenuConfig(0xF7), length: 0x02, data: 02 00 00` and added an answer to these requests, maybe it works. Please check.

Should I do that? How? :)

**Reply by MichaelDvP** [reply]
> Should I do that? How? :)

I've updated my build with (a fix for) the reply, just update and make a log to see the reply and how the controller react on this. I think next there will be a version request and hopefully then the controller is happy and do not request the remote thermostat anymore.

**Reply by mattfro** [reply]
Oh then I have it...
this from yesterday?
![image](https://github.com/emsesp/EMS-ESP32/assets/46936922/cac50360-0e90-4477-ab8b-3d652b7e7c15)

[rc200_testbuild8.log](https://github.com/emsesp/EMS-ESP32/files/14993701/rc200_testbuild8.log)

**Reply by MichaelDvP** [reply]
Ok, i did not see the F7 requests anymore in this log, so it seems to work. The first request with reply is in the first part without full log.

I've found now the cause for the incomplete telegrams, i did not know that the master also polls adresses with MSB set.
My new build should work now without extra incomplete messages.

Could you make a log with setting `log all` before sending the first remotetemp, to get the initial communication in the log.

**Reply by mattfro** [reply]
Here you go....I hope I got everything.
[rc200_testbuild9.log](https://github.com/emsesp/EMS-ESP32/files/15016901/rc200_testbuild9.log)

**Reply by MichaelDvP** [reply]
Ah, misunderstanding, i like to see a minute before activating the remote until a minute after activting to see what the controller sends/queries to/from the remote. It's not important, i know from testbuid8.log, that the F7 requests are gone and i guess that the contorller first requests F7 for telegram 02, remote replies, controller requests 02 telegram, remote replies, after that no more requests from controller.

**Reply by mattfro** [reply]
[rc200_before_after.log](https://github.com/emsesp/EMS-ESP32/files/15023135/rc200_before_after.log)
here I hope this is better :)

**Reply by MichaelDvP** [reply]
> here I hope this is better :)

Yes, but still not what i expected, i give up. Seems the contoller have all info he wants and just don't ask again. It's not worth reset all and start again, the main info is: the repeated requests are gone an no more incompletes from the remote.

The few incompletes that are still there (before and after activation of remote) makes me wonder.
```
ems-esp:$ 000+00:01:07.971 T 170: [telegram] Incomplete Rx: 8B 90 FF 00 19 01 FA
ems-esp:$ 000+00:02:07.991 T 504: [telegram] Incomplete Rx: 8B 90 FF 00 19 01 FD
ems-esp:$ 000+00:02:08.991 T 509: [telegram] Incomplete Rx: 8B 90 FF 00 19 03 EE
ems-esp:$ 000+00:03:06.991 T 877: [telegram] Incomplete Rx: 8B 88 FF 18 19 03 F1
ems-esp:$ 000+00:03:23.991 T 983: [telegram] Incomplete Rx: 8B 90 FF 00 19 01 F2
```

Looks all as valid requests from device 8B to thermostat and boiler. But incomplete means the crc is wrong. 
We have no device 8B, but ems-esp is 0B, a wrong MSB would explain the wrong crc. 
But the telegrams requested are all unknown in ems-esp, we never ask for 01FA, 10FD, 03EE, 03F1 or 01F2. 
Very strange, @proddy Any explaination for these telegrams?
It's nothing to worry about, but very interesting.

**Reply by proddy** [reply]
must be something special we don't know about. Maybe having the src's MSB set (8B) means something special we don't know about yet, like it's pretending to be another device. We need to see the CRC codes and then check if the whole telegram is still valid without the MSB set. So a special build with a modified `RxService::add()` that also shows the CRC final byte to test?

**Reply by mattfro** [reply]
Internet module, or maybe a pump(circulation pump?). Not sure what else there is... circulation pump would most likely not talk to a thermostat? :) Then there is the outdoor temperature sensor... 

did a deep scan, not sure what does that tell us?

boiler: Bosch Enviline/Compress 6000AW/Hybrid 3000-7000iAW/SupraEco/Geo 5xx/WLW196i/WSW196i (DeviceID:0x08, ProductID:172, Version:03.01)
 Received telegram type IDs: 0xBF 0xC2 0xD1 0xE3 0xE4 0xE5 0xE9 0x494 0x495 0x48F 
 Fetched telegram type IDs: 0x14 0xE6 0xEA 0x48D 0x48A 0x4A2 0x485 0x492 0x488 0x484 0x48B 0x491 0x499 0x49C 0x49D 0x2CC 0x4A5 
 Pending telegram type IDs: 0x10 0x11 0x15 0x1C 0x18 0x19 0x1A 0x35 0x16 0x33 0x34 0x26 0x2A 0x28 0x486 0x4AE 0x4AF 
 Ignored telegram type IDs: 0xF7 0x4A0 0x4A3 0x4B4 0x497 0x49F 

thermostat: Rego 2000/3000 (DeviceID:0x10, ProductID:172, Version:03.01)
 Received telegram type IDs: 0x06 0x31D 
 Fetched telegram type IDs: 0x2A5 0x2B9 0x2AF 0x29B 0x471 0x2A6 0x2BA 0x2B0 0x29C 0x472 0x467 0x291 0x468 0x292 0x2F5 0x23A 0x240 
 Pending telegram type IDs: 0xA3 0xA2 0x12 0x13 0x2A7 0x2BB 0x2B1 0x29D 0x473 0x2A8 0x2BC 0x2B2 0x29E 0x474 0x2A9 0x2BD 0x2B3 0x29F 0x475 0x2AA 0x2BE 0x2B4 0x2A0 0x476 0x2AB 0x2BF 0x2B5 0x2A1 0x477 0x2AC 0x2C0 0x2B6 0x2A2 0x478 0x2CC 0x2CE 0x2D0 0x469 0x293 0x2D2 0x46A 0x294 0x2F6 0x31B 0x31E 0x267 0xBB 0x23E 
 Ignored telegram type IDs: 0x42B 0xF7 

thermostat: RC200/CW100 (DeviceID:0x38, ProductID:157, Version:32.02)
 Received telegram type IDs: 0x42B 
 Fetched telegram type IDs: 
 Pending telegram type IDs: 
 Ignored telegram type IDs: 

thermostat: Generic thermostat (DeviceID:0x17)
 Received telegram type IDs: 
 Fetched telegram type IDs: 
 Pending telegram type IDs: 0xA3 0x06 0xA2 0x12 0x13 0xB1 0xB0 
 Ignored telegram type IDs: 

thermostat: Generic thermostat (DeviceID:0x19)
 Received telegram type IDs: 
 Fetched telegram type IDs: 
 Pending telegram type IDs: 0xA3 0x06 0xA2 0x12 0x13 0xB1 0xB0 
 Ignored telegram type IDs: 

thermostat: RF room temperature sensor (DeviceID:0x40)
 Received telegram type IDs: 
 Fetched telegram type IDs: 
 Pending telegram type IDs: 0x435 
 Ignored telegram type IDs: 

gateway: KM200/MB LAN 2 (DeviceID:0x48, ProductID:189, Version:04.08)

controller: Rego 3000 (DeviceID:0x09, ProductID:240, Version:38.03)
 Received telegram type IDs: 0x06 
 Fetched telegram type IDs: 
 Pending telegram type IDs: 
 Ignored telegram type IDs: 0xC6 0xC7 0xBF

**Reply by MichaelDvP** [reply]
I've taken a deeper look, it is really a scrambled ems-esp request with first and last byte disturbed and, because it is not answered, the ems-esp repeats the request direct after this. So we do not miss any data.
e.g. `Incomplete Rx: 8B 88 FF 18 19 03 F1` should be `0B 88 FF 18 19 03 8B` => Me(0x0B) -R-> boiler(0x08), HPPumps(0x048B), length: 0x19 (offset 24)
(This message with offset gives me the hint, but all others follow the same scheme)
My theory is, that the first byte was interupted and stopbit is shifted to MSB, the master has polled 0B, but did not get an ack from 0B (receives 8B), so timeouts and send next poll. This overwrites the last (7th) byte of our request.
> So a special build with a modified RxService::add() that also shows the CRC final byte to test?

We always show the crc for raw messages and skip in formatted telegram display, don't know why we skip it in this log-message, we should just show the full raw telegram here as in trace-raw or watch-raw.
> did a deep scan, not sure what does that tell us?

deep scan is imho useless and needs some rework. Since we generate empty replys after 3 repeats and have generic types for detecting tado thermostats, it gives wrong results.

**Reply by mattfro** [reply]
@MichaelDvP  added new logs from your latest build
[rc200_before_after2.log](https://github.com/emsesp/EMS-ESP32/files/15036849/rc200_before_after2.log)

**Reply by mattfro** [reply]
and same with just watch raw
[rc200_before_after2_raw.log](https://github.com/emsesp/EMS-ESP32/files/15037688/rc200_before_after2_raw.log)

**Reply by mattfro** [reply]
Adding also combo log all and watch raw to make it more easier maybe :)
[rc200_before_after2_combo.log](https://github.com/emsesp/EMS-ESP32/files/15037744/rc200_before_after2_combo.log)

**Reply by proddy** [reply]
@MichaelDvP to show the CRC in the log we would need to include it the in the Telegram class. It's shown in the console with `watch raw` since its pre-processing. I agree 'deep scan' is stupid and should be removed.

**Reply by MichaelDvP** [reply]
> to show the CRC in the log we would need to include it the in the Telegram class

Yes, it's only the change i've made here: https://github.com/MichaelDvP/EMS-ESP32/commit/037175f7b022dde3855232b2dd80f60ada6d0b8d

**Reply by mattfro** [reply]
I used that for the latest logs....

**Reply by MichaelDvP** [reply]
Yes, i've seen it. The raw also helped to remember, that your system uses HT3 protocoll, so the `8B` is correct and only the last byte is broken. 
```
ems-esp:$ 000+00:01:13.372 T 1010: [telegram] Incomplete Rx: 8B 90 FF 00 19 01 FF A5
ems-esp:$ 000+00:01:14.623 N 1011: [telegram] Rx: 8B 90 FF 00 19 01 F6 A5
``` 
first with broken last byte, second as resend from ems-esp correct, both transmitted with identical crc `A5`.

**Reply by mattfro** [reply]
Hmm... ht3, does it mean I need to set that instead of ems or ems+ in ems-esp?
Because, if I use ht3, it will miss a lot of entities and remotetemp for example will not work.
HT3:
![image](https://github.com/emsesp/EMS-ESP32/assets/46936922/35430bb3-eb02-48c8-a16a-f4efd399de91)
EMS+:
![image](https://github.com/emsesp/EMS-ESP32/assets/46936922/10de08ac-f5ae-4a87-bbc5-ee4cfd369c1d)


Or is this something you kind of handle differently from the settings you have in ems-esp?

**Reply by MichaelDvP** [reply]
> ht3, does it mean I need to set that instead of ems or ems+ in ems-esp?

No. Protocol and timing  are different things, you pick the tx-mode that works best. 

HT3 is the Junkers electronic and HT3 protocol is the same as EMS, only in first byte (sender) the highest bit is set. Each telegram we send starts with 0B in a buderus system, but with 8B in Junkers. This makes the devicesof different brands  incompatble. EMS-ESP checks the bus on start and select the right protocol automatically. Also Junkers uses different telegram numbers/orders. The Bosch labled modules uses the same telegram numbers as buderus, but addressing like Junkers, so also incompatible, you can't connect Junkers or Buderus modules to a Bosch heating.

Tx-mode is the send timing: The client devices send by current modulation, the master by voltage modulation. This allows full duplex (Hardware mode), but  depending on line impedance drawing current also influences the voltage. When sending the master repeats every byte sendet bythe device to publish it to the other devices, In tx-mode EMS we wait for the master byte before sending the next. The older Junkers seems to have a lower timeout, we need to start the next byte before the master echo is completed (HT3). EMS+ is less critical and we can wait a bit longer than one byte to allow voltage to be stabilized after the sending.

**Reply by proddy** [reply]
I'm adding that to the wiki Michael!

**Reply by mattfro** [reply]
Thanks for the explanation 🙂 last question… is this done now or something still missing? I mean it works but did you want to test something still?  Otherwise we can close this discussion I guess?

**Reply by MichaelDvP** [reply]
I think it's all done, we can close this discussion. Thanks for all the logs.

**Comment by mattfro** [comment]
@MichaelDvP  seems remotetemp is gone... I can't see that anymore in latest ems-esp 3.7 dev10 version.
![image](https://github.com/emsesp/EMS-ESP32/assets/46936922/4e34e6d3-291a-43dc-9526-ec3134087193)
it's hidden in customization
![image](https://github.com/emsesp/EMS-ESP32/assets/46936922/a8d9ad66-c52d-440f-9364-d843157a0bee)
so I can't set it at the moment... some change in later dev that has caused this?

**Reply by MichaelDvP** [reply]
There is no change, this was and is intended, the temp shows up if set, and is hidden when unset. Command is still active.

**Reply by mattfro** [reply]
hmm...so what should I set then I remotetemp can't be set? I used that before to set the temperature, now it's not showed in hass.
Aah, I think I understand now, need to use restcommand to set it?
service: rest_command.emsesp

so can't use mqtt to set it... because it's not there before it's set :)

**Reply by mattfro** [reply]
Sorry my mistake or misunderstanding... but in a way it would be good if that entity could be sent via mqtt even if it's empty just for this usecase? or maybe stupid for a "small" thing, have to make changes in the code.

**Reply by proddy** [reply]
I think when we moved the DeviceType to CMD it got removed from MQTT Discovery - so we need to extend support for creation of discovery topics also for Commands. I think. Haven't checked the code.

**Reply by MichaelDvP** [reply]
@mattfro  You can always use mqtt/api/telnet to set the value. It's a value to use in automation, not designed to show up web/HA if not set.

**Reply by mattfro** [reply]
hmm ok... I see. Well it works via api from an automation. So let's close this again.
I had it first as the entity in homeasisstant, but when I clean it up that entity was gone. :) So that why I was wondering is it broken :)
Anyways it works with api via automation. So we can close this again. Thank you again :)

**Comment by per-lind** [comment]
@mattfro did you ever get this working in a stable way without errors?

**Reply by mattfro** [reply]
Hi, yes it works without any issues. 
But you need to use dev version, I think it's only in 3.7.0 dev still
